src/Compilers/VisualBasic/Portable/Lowering/SyntheticBoundNodeFactory.vb

```diff
@@ -851,31 +855,58 @@ nextm:
+        Public Function MethodInfo(meth As WellKnownMember, systemReflectionMethodInfo As TypeSymbol) As BoundExpression
             Dim method = WellKnownMember(Of MethodSymbol)(meth)
             If method Is Nothing Then
                 Return BadExpression()
             Else
-                Return MethodInfo(method)
+                Return MethodInfo(method, systemReflectionMethodInfo)
             End If
         End Function
 
-        Public Function MethodInfo(meth As SpecialMember) As BoundExpression
+        Public Function MethodInfo(meth As SpecialMember, systemReflectionMethodInfo As TypeSymbol) As BoundExpression
             Dim method = DirectCast(SpecialMember(meth), MethodSymbol)
             If method Is Nothing Then
                 Return BadExpression()
             Else
-                Return MethodInfo(method)
+                Return MethodInfo(method, systemReflectionMethodInfo)
             End If
         End Function
 
-        Public Function MethodInfo(method As MethodSymbol) As BoundExpression
-            Dim boundNode = New BoundMethodInfo(Syntax, method, WellKnownType(Microsoft.CodeAnalysis.WellKnownType.System_Reflection_MethodInfo))
-            ' Touch the method to be used to report use site diagnostics
-            WellKnownMember(Of MethodSymbol)(If(method.ContainingType.IsGenericType,
-                                                Microsoft.CodeAnalysis.WellKnownMember.System_Reflection_MethodBase__GetMethodFromHandle2,
-                                                Microsoft.CodeAnalysis.WellKnownMember.System_Reflection_MethodBase__GetMethodFromHandle))
+        Public Function MethodInfo(method As MethodSymbol, systemReflectionMethodInfo As TypeSymbol) As BoundExpression
+            Debug.Assert(systemReflectionMethodInfo.Equals(Compilation.GetSpecialType(InternalSpecialType.System_Reflection_MethodInfo), TypeCompareKind.AllIgnoreOptionsForVB) OrElse
+                         systemReflectionMethodInfo.Equals(Compilation.GetWellKnownType(CodeAnalysis.WellKnownType.System_Reflection_MethodInfo), TypeCompareKind.AllIgnoreOptionsForVB) OrElse
+                         systemReflectionMethodInfo.Equals(Compilation.GetWellKnownType(CodeAnalysis.WellKnownType.System_Reflection_ConstructorInfo), TypeCompareKind.AllIgnoreOptionsForVB))
+
+            Dim getMethodFromHandle As MethodSymbol
+            Dim isInGenericType As Boolean = DirectCast(method.ContainingType.GetTupleUnderlyingTypeOrSelf(), NamedTypeSymbol).IsGenericType OrElse method.ContainingType.IsAnonymousType
+
+            If systemReflectionMethodInfo.ExtendedSpecialType = InternalSpecialType.System_Reflection_MethodInfo Then
+                getMethodFromHandle = DirectCast(SpecialMember(
+                                                     If(isInGenericType,
+                                                         Microsoft.CodeAnalysis.SpecialMember.System_Reflection_MethodBase__GetMethodFromHandle2,
```

Comment:
>Microsoft. [](http://example.com/codeflow?start=57&length=10)

nit: looks like `Microsoft.` part can be simplified

Reply:
> nit: looks like `Microsoft.` part can be simplified

Things like that do not bother me a bit.